### PR TITLE
[Overlay ECMP] API to set and get BGP community string

### PR DESCRIPTION
**Pull request by sumukhatv**

### CODE REVIEW

File: go-server-server/go/default.go [hunk 1]
```diff
@@ -67,6 +67,51 @@ func ConfigResetStatusPost(w http.ResponseWriter, r *http.Request) {
...
 }
 
+func ConfigBgpProfilePost(w http.ResponseWriter, r *http.Request) {
+    var attr BgpProfileModel
+    vars := mux.Vars(r)
+    db := &app_db_ops
+
+    ReadJSONBody(w, r, &attr)
+
+    pt := swsscommon.NewProducerStateTable(db.swss_db, BGP_PROFILE)
```
Comment (sumukhatv) [inline go-server-server/go/default.go:77]: Change this to normal Table

File: go-server-server/go/default.go [hunk 2]
```diff
@@ -67,6 +67,51 @@ func ConfigResetStatusPost(w http.ResponseWriter, r *http.Request) {
...
+    w.WriteHeader(http.StatusNoContent)   
+}
+
+func ConfigBgpProfileGet(w http.ResponseWriter, r *http.Request) {
+    vars := mux.Vars(r)
+    db := &app_db_ops
+
+    var bgp_profile_tb string
+
+    if *RunApiAsLocalTestDocker {
```
Comment (sumukhatv) [inline go-server-server/go/default.go:92]: No need to check for this

File: go-server-server/go/models.go [hunk 3]
```diff
@@ -18,6 +18,10 @@ type ConfigResetStatusModel struct {
     ResetStatus      string `json:"reset_status,omitempty"`
 }
 
+type BgpProfileModel struct {
```
Comment (prsunny) [inline go-server-server/go/models.go:21]: Please add this to sonic-api.yml file

File: go-server-server/go/default.go [hunk 4]
```diff
@@ -67,6 +67,43 @@ func ConfigResetStatusPost(w http.ResponseWriter, r *http.Request) {
...
+    vars := mux.Vars(r)
+    db := &app_db_ops
+
+    ReadJSONBody(w, r, &attr)
+
+    bgp_profile_t := swsscommon.NewTable(db.swss_db, BGP_PROFILE_TABLE)
+    defer bgp_profile_t.Delete()
+
+    bgp_profile_t.Set(vars["profile_name"], map[string]string {
+                "community": attr.Community,
```
Comment (prsunny) [inline go-server-server/go/default.go:81]: community_id

File: sonic_api.yaml [hunk 5]
```diff
@@ -249,6 +249,74 @@ paths:
...
+        - name: profile_name
+          in: path
+          required: true
+          type: string
+          description: BGP profile name
+        - name: community_id
+          in: path
+          required: true
+          type: string
+          description: Community ID for the BGP profile
```
Comment (prsunny) [inline sonic_api.yaml:301]: Can you please clarify the format as "16bit:16bit" - Eg. "1234:1235"

File: go-server-server/go/default.go [hunk 6]
```diff
@@ -67,6 +67,43 @@ func ConfigResetStatusPost(w http.ResponseWriter, r *http.Request) {
...
+    }
+
+    if kv["community_id"] == "" {
+        WriteRequestError(w, http.StatusBadRequest, "Malformed arguments for API call", []string{"profile_name"}, "Invalid profile_name")
+        return
+    }
+
+    output := BgpProfileModel {
+            CommunityId: kv["community_id"],
+        }
```
Comment (prsunny) [inline go-server-server/go/default.go:103]: Please fix indentation

File: go-server-server/go/default.go [hunk 7]
```diff
@@ -67,6 +67,43 @@ func ConfigResetStatusPost(w http.ResponseWriter, r *http.Request) {
...
+    db := &app_db_ops
+
+    ReadJSONBody(w, r, &attr)
+
+    bgp_profile_t := swsscommon.NewTable(db.swss_db, BGP_PROFILE_TABLE)
+    defer bgp_profile_t.Delete()
+
+    bgp_profile_t.Set(vars["profile_name"], map[string]string {
+                        "community_id": attr.CommunityId,
+        }, "SET", "")
```
Comment (prsunny) [inline go-server-server/go/default.go:82]: where do we handle delete?

**Review by prsunny — APPROVED** [review]

**Pull request merged**